### PR TITLE
Distant models: handle endpoint parameters

### PR DESCRIPTION
**Pull request by antoinejeannot**
This PR aims at handling endpoint parameters by distant models, parameters that should be defined at the Model's instantiation or at predict time.

Looking forward to your feedback on this.
Thanks!

### CODE REVIEW

File: modelkit/core/models/distant_model.py [hunk 1]
```diff
@@ -50,17 +57,25 @@ class AsyncDistantHTTPModel(AsyncModel[ItemType, ReturnType]):
...
             self.aiohttp_session = aiohttp.ClientSession()
+        endpoint_params = kwargs.get("endpoint_params")
+        built_endpoint = (
+            self.built_endpoint
+            if not endpoint_params
+            else build_url(self.endpoint, endpoint_params)
+        )
         async with self.aiohttp_session.post(
-            self.endpoint,
+            built_endpoint,
```
Comment (ldeflandre) [inline modelkit/core/models/distant_model.py:78]: We can save us the custom build_url logic if we use `params` argument here, can't we ?

**Comment by victorbenichoux** [comment]
Should we have these set via the kw args of the predict?

**Comment by antoinejeannot** [comment]
@victorbenichoux good point! I would prefer consistency using `endpoint_params` in both `model_settings`  and `**kwargs` parameters, instead of passing everything from the `**kwargs`, hence this implementation.

**Review by victorbenichoux — APPROVED** [review]

File: modelkit/core/models/distant_model.py [hunk 2]
```diff
@@ -79,17 +81,19 @@ class DistantHTTPModel(Model[ItemType, ReturnType]):
...
         pass
 
     @retry(**SERVICE_MODEL_RETRY_POLICY)
-    def _predict(self, item):
+    def _predict(self, item, **kwargs):
         if not self.requests_session:
             self.requests_session = requests.Session()
         response = self.requests_session.post(
             self.endpoint,
+            params=kwargs.get("endpoint_params", self.endpoint_params),
```
[the remark this answers is not in the document]
Comment (antoinejeannot) [reply modelkit/core/models/distant_model.py:96]: I have hesitated ahah `endpoint_params` seems more explicit (to me): the params are for the `endpoint` url, not for the model predict.

[the remark this answers is not in the document]
Comment (antoinejeannot) [reply modelkit/core/models/distant_model.py:96]: I'll let you decide @victorbenichoux

**Pull request merged**